baselines/fedexp/README.md

```diff
@@ -0,0 +1,139 @@
+
+
+****Training Hyperparameters:**** :
+The following tables show the main hyperparameters for this baseline with their default value (i.e. the value used if you run `python main.py` directly)
+
+| Description                 | Default Value                       |
+|-----------------------------|-------------------------------------|
+| total clients               | 100                                 |
+| clients per round           | 20                                  |
+| number of rounds            | 500                                 |
+| number of local rounds      | 2                                   |
+| batch_size                  | 50                                  |
+| client resources            | {'num_cpus': 2.0, 'num_gpus': 0.1 } |
+| eta_l (local learning rate) | 0.01                                |
+| epsilon                     | 0.001                               |
+| decay                       | 0.998                               |
+| weight_decay                | 0.0001                              |
+| max_norm                    | 10                                  |
+
+For Dataset:
+Choice of alpha parameter for the Dirichlet distribution used to create heterogeneity in the client datasets for CIFAR
+
+| Description | Default Value |
+|-------------|---------------|
+| alpha       | 0.3           |
+
+## Environment Setup
+
+To construct the Python environment follow these steps:
+
+```bash
+# set local python version via pyenv
+pyenv local 3.10.6
+
+# then fix that for poetry
+poetry env use 3.10.6
+
+# then install poetry env
+poetry install
+
+# activate the environment
+poetry shell
+
+```
+
+## Running the Experiments
+
+To run this FedExP with CIFAR10 baseline, first ensure you have activated your Poetry environment (execute `poetry shell` from this directory), then:
+```bash  
+python -m fedexp.main # this will run using the default settings in the `conf/config.yaml`
+
+# you can override settings directly from the command line
+python -m fedexp.main seed=33 num_rounds=200 # will set seed to 33 and the number of rounds to 200
+
+# if you run this baseline with a larger model, you might want to use the GPU (not used by default).
+# you can enable this by overriding the `server_device` and `client_resources` config. For example
+# the below will run the server model on the GPU and 20 clients will be allowed to run concurrently on a GPU (assuming you also meet the CPU criteria for clients)
+python -m fedprox.main server_device=cuda client_resources.num_gpus=0.01
+```
+
+To run using CIFAR100:
+
+```bash
+poetry run -m fedexp.main --config-name cifar100
+```
+
+To run using FedAvg:
+
+```bash
+poetry run -m fedexp.main strategy.algorithm=fedavg
+```
+
+
+## Expected Results
+
+```bash
+poetry run -m fedexp.main --multirun strategy.alogrithm=[fedexp,fedavg]
+```
+| FedExP | FedAvg |
+|:----:|:----:|
+|![FedExP CIFAR10](_static/FedExP_Cifar10_S0.png) | ![FedAvg CIFAR10](_static/FedAvg_Cifar10_S5.png) |
```

Comment:
Better to show these two in a single plot, so they are easier to compare. Please use grid lines and a title to the plot.

Reply:
Same comment for CIFAR-100